vortex-array/src/children.rs

```diff
@@ -25,6 +30,18 @@ impl NamedChildrenCollector {
     }
 }
 
+impl NamedTreeCollector {
```

Comment:
we never use the names from this collector. We just collect all the children

Reply:
Want me to remove?

Reply:
I'm just confused how it's related to this pr. I know we want it for some display and transformations but it's a bit out of place in this pr

Reply:
Its gone, I needed to get the encoding of all arrays and I didn't see the other iterator